### ISSUE TITLE
 test the microdrive board

### ISSUE THREAD

**Issue by dvogureckiy99** [open-dynamic-robot-initiative/udriver_firmware#8]
Are there code for microdrive for calibrating motors?
Are there code that I could somehow test the microdrive board ?

**Comment by dvogureckiy99** [comment]
Thank Thomas Flayols, switching two phases on the motor helped me. But whether I reverse phase order programmaticly?

**Comment by dvogureckiy99** [comment]
> Hello, I'm not sure if I understand your question correctly.  Are you asking if it is possible to switch the motor phases in the software?  As far as I know there is know option for this, the phases have to be connected in the right order physically.

Yes, I already understood it,but on many of an other boards, like Odrive, there is exist this functionality  Maybe there is possibility to implement it on this board?

**Comment by thomasfla** [comment]
Hi,
Sorry, we did not implement phases switch programmatically. 
Here is an element of why:
The way we communicate with the boards is by exchanging a fixed shape, compact and fast SPI bidirectional transaction. We don't have a flexible way to reprogram the settings of the board, such as: motor R and L, number of pole pairs, phase order etc..
Instead, those parameters are set at compile time before flashing the boards. 
This motor board was intended to be tight to a motor type, and is not as generic as Odrive for example.
On the plus side, we have a very well optimized communication to do real time control at high frequency. 

We are developing a new electronic (OMODRI) that will make life easier to implement other communication buses for configuration etc. 

We have no plan to implement this features on this electronic. We think that identifying once the phase order  and marking the wire is not a huge constrained for ODRI robot at this time.

**Comment by dvogureckiy99** [comment]
But I saw that there is possibility to set up boards parameters via CAN, is it real?
